app/controllers/HashtagContentController.java

```diff
@@ -2,17 +2,22 @@
 
 import com.google.inject.Inject;
 import play.mvc.*;
-import services.dataAccess.AbstractDataAccess;
 import services.dataAccess.RedisAccessObject;
-import services.dataAccess.InMemoryAccessObject;
+import services.dataAccess.proto.PostListProto.PostList;
+import services.serializer.BinarySerializer;
+
+import java.util.Optional;
+
 /**
  * This controller contains an action to handle HTTP requests
  * to the application's home page.
  */
 public class HashtagContentController extends Controller {
 
     @Inject
-    private InMemoryAccessObject dataSource = new InMemoryAccessObject();
+    private RedisAccessObject dataSource = new RedisAccessObject();
```

Comment:
private AbstractDataAccess dataSource = new RedisAccessObject();

Assign realisations to general types so that we know we can switch it out easily and there is no dependence on RedisAccessObject for this class to work.